GPU/Common/TextureDecoder.h

```diff
@@ -188,35 +188,23 @@ inline void DeIndexTexture4(ClutT *dest, const u8 *indexed, int length, const Cl
 	}
 }
 
-template <typename ClutT>
-inline void DeIndexTexture4Optimal(ClutT *dest, const u8 *indexed, int length, ClutT color) {
-	for (int i = 0; i < length; i += 2) {
-		u8 index = *indexed++;
-		dest[i + 0] = color | ((index >> 0) & 0xf);
-		dest[i + 1] = color | ((index >> 4) & 0xf);
-	}
-}
-
-template <>
-inline void DeIndexTexture4Optimal<u16>(u16 *dest, const u8 *indexed, int length, u16 color) {
+inline void DeIndexTexture4Optimal(u16 *dest, const u8 *indexed, int length, u16 color) {
 	const u16_le *indexed16 = (const u16_le *)indexed;
-	const u32 color32 = (color << 16) | color;
-	u32 *dest32 = (u32 *)dest;
-	for (int i = 0; i < length / 2; i += 2) {
-		u16 index = *indexed16++;
-		dest32[i + 0] = color32 | ((index & 0x00f0) << 12) | ((index & 0x000f) >> 0);
-		dest32[i + 1] = color32 | ((index & 0xf000) <<  4) | ((index & 0x0f00) >> 8);
+	const u64 color64 = ((u64)color << 48) | ((u64)color << 32) | ((u64)color << 16) | (u64)color;
+	u64_le *dest64 = (u64_le *)dest;
+	for (int i = 0; i < length / 4; i ++) {
+		u64 index = *indexed16++;
+		dest64[i] = color64 | ((index & 0xf000) <<  36) | ((index & 0x0f00) << 24) | ((index & 0x00f0) << 12) | ((index & 0x000f) << 0);
 	}
 }
 
 inline void DeIndexTexture4OptimalRev(u16 *dest, const u8 *indexed, int length, u16 color) {
 	const u16_le *indexed16 = (const u16_le *)indexed;
-	const u32 color32 = (color << 16) | color;
-	u32 *dest32 = (u32 *)dest;
-	for (int i = 0; i < length / 2; i += 2) {
-		u16 index = *indexed16++;
-		dest32[i + 0] = color32 | ((index & 0x00f0) << 24) | ((index & 0x000f) << 12);
-		dest32[i + 1] = color32 | ((index & 0xf000) << 16) | ((index & 0x0f00) <<  4);
+	const u64 color64 = ((u64)color << 48) | ((u64)color << 32) | ((u64)color << 16) | (u64)color;
+	u64_le *dest64 = (u64_le *)dest;
```

Comment:
we can safely assume 8-byte alignment here right ?

Reply:
I believe so.

-[Unknown]